network/src/peers/peer/cipher.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::convert::TryInto;
+
+use snow::TransportState;
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+
+use crate::NetworkError;
+
+pub struct Cipher {
+    state: TransportState,
+    buffer: Box<[u8]>,
+    noise_buffer: Box<[u8]>,
+}
+
+impl Cipher {
+    pub fn new(state: TransportState, buffer: Box<[u8]>, noise_buffer: Box<[u8]>) -> Self {
+        assert_eq!(buffer.len(), crate::MAX_MESSAGE_SIZE);
+        assert_eq!(noise_buffer.len(), crate::MAX_MESSAGE_SIZE);
```

Comment:
the `noise` buffer doesn't need to be larger than `65535B`, as that's the max message limit in the specification